apps/email/js/cards/compose.js

```diff
@@ -361,6 +361,18 @@ return [
       };
     },
 
+    haveAddressesChanged: function() {
+      var addrs = this.extractAddresses();
+      function compareAddresses(a, b) {
+        return a.length == b.length && !a.some(function(e, i) {
+          return typeof b[i].address === 'undefined' || e !== b[i].address;
```

Comment:
*...does this function work in that case?*
I have tested it and it seems to work in the same way as working in the version without my corrections, before I have started working on it.


*If it turns out it is best for all addresses coming into this card to be of form {name: ..., address: ...}, then we could look at just creating that object without  the name part in the message_reader change where it does the model.api.parseMailbox(addr).address normalization.*
I do not understand what do you mean. Could you describe it in another way?